### ISSUE TITLE
WinUI 3 Single Project MSIX crashes when built for x86

### ISSUE THREAD

**Issue by mattleibow** [microsoft/WindowsAppSDK#2064]
### Describe the bug

I have a single project app that crashes on launch with a `BadImageFormatException`:

```
System.BadImageFormatException: 'An attempt was made to load a program with an incorrect format. (0x8007000B)'
This exception was originally thrown at this call stack:
    App1.Program.Main(string[]) in App.g.i.cs
```

### Steps to reproduce the bug

1. Download and extract this project: [App1.zip](https://github.com/microsoft/WindowsAppSDK/files/7998321/App1.zip)
2. Hit F5 / Run
3. Observe exception
4. Change the `<RuntimeIdentifier>win10-x86</RuntimeIdentifier>` to be `win10-x64`
5. Hit F5 / Run
6. Observe NO exception

I created this by making a new WinUI 3 app, converted it to a single project and then make the app be an AnyCPU with a x86 RID.

### Expected behavior

No crash - like `win10-x64`.

### Screenshots

![image](https://user-images.githubusercontent.com/1096616/152431469-0e187eac-9f94-4356-9d6b-d68227d0bb39.png)

### NuGet package version

1.0.0

### Packaging type

Packaged (MSIX)

### Windows version

Windows 11 version 21H2 (22000)

### IDE

Visual Studio 2022-preview

AzDo bug: https://dev.azure.com/microsoft/OS/_workitems/edit/38094973

**Comment by kmgallahan** [comment]
Works fine if you use the single project template instead of converting (or rather doing whatever process resulted in the solution you provided:

![Screenshot 2022-02-03 214505](https://user-images.githubusercontent.com/5379488/152463982-e06268c7-4d83-4648-9708-a12de644e0c2.jpg)

csproj looks a bit different (note I updated the sdk build tools from the version that appears in the template).

```
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>WinExe</OutputType>
    <TargetFramework>net6.0-windows10.0.19041.0</TargetFramework>
    <TargetPlatformMinVersion>10.0.17763.0</TargetPlatformMinVersion>
    <RootNamespace>App7</RootNamespace>
    <ApplicationManifest>app.manifest</ApplicationManifest>
    <Platforms>x86</Platforms>
    <RuntimeIdentifiers>win10-x86</RuntimeIdentifiers>
    <PublishProfile>win10-$(Platform).pubxml</PublishProfile>
    <UseWinUI>true</UseWinUI>
    <EnablePreviewMsixTooling>true</EnablePreviewMsixTooling>
  </PropertyGroup>

  <ItemGroup>
    <Content Include="Assets\SplashScreen.scale-200.png" />
    <Content Include="Assets\LockScreenLogo.scale-200.png" />
    <Content Include="Assets\Square150x150Logo.scale-200.png" />
    <Content Include="Assets\Square44x44Logo.scale-200.png" />
    <Content Include="Assets\Square44x44Logo.targetsize-24_altform-unplated.png" />
    <Content Include="Assets\StoreLogo.png" />
    <Content Include="Assets\Wide310x150Logo.scale-200.png" />
  </ItemGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.WindowsAppSDK" Version="1.0.0" />
    <PackageReference Include="Microsoft.Windows.SDK.BuildTools" Version="10.0.22543-preview" />
    <Manifest Include="$(ApplicationManifest)" />
  </ItemGroup>

  <!-- Defining the "Msix" ProjectCapability here allows the Single-project MSIX Packaging
       Tools extension to be activated for this project even if the Windows App SDK Nuget
       package has not yet been restored -->
  <ItemGroup Condition="'$(DisableMsixProjectCapabilityAddedByProject)'!='true' and '$(EnablePreviewMsixTooling)'=='true'">
    <ProjectCapability Include="Msix" />
  </ItemGroup>
</Project>
```

**Comment by mattleibow** [comment]
I believe it is because you have included the `<Platforms>x86</Platforms>` property. Since we are building a multi-targeting project, we cannot use that because all the projects use `AnyCPU`.

**Comment by kmgallahan** [comment]
I didn't think you could use anycpu with WinUI:

https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/stable-channel#other-limitations-and-known-issues

No support for Any CPU build configuration: When [adding the Windows App SDK](https://docs.microsoft.com/en-us/windows/apps/windows-app-sdk/use-windows-app-sdk-in-existing-project) to an existing .NET application or component that supports Any CPU, you must specify the desired architecture: x86, x64 or arm64.

**Comment by mattleibow** [comment]
You can in most cases (for example if you target x64), but we need to. In a multitargeting world with a single project in .NET Maui, this is basically the only way. 



Partially related to this comment: 
 - https://github.com/microsoft/WindowsAppSDK/issues/893
 - https://github.com/microsoft/WindowsAppSDK/issues/1217

**Comment by Scottj1s** [comment]
A fix for this issue will be available in Windows App SDK 1.0.2 and 1.1
